# MIT License
#
# Copyright (c) 2025 Meher V.R. Malladi
#
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice shall be included in all
# copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
# SOFTWARE.
imu_topic: /imu_raw
imu_frame: imu_link
lidar_topic: /points_raw
lidar_frame: velodyne
base_frame: velodyne
odom_frame: odom_lio
odom_topic: /rko_lio/odometry
invert_odom_tf: false
publish_local_map: true
map_topic: /rko_lio/local_map
# extrinsics for lidar base frame
extrinsic_lidar2base: [0, 0, 0, 1, 0, 0, 0]
extrinsic_imu2base: [0.0, 0.0, 0.0, 1.0, 0.05, 0.0, -0.12]
# lio config
deskew: true
voxel_size: 1.0
max_points_per_voxel: 20
max_range: 100.0
min_range: 1.0
max_correspondance_distance: 0.5
convergence_criterion: 1e-4
initialization_phase: true
max_expected_jerk: 3.0
double_downsample: true
min_beta: 200
